docs/datasync/soft-delete.md

```diff
@@ -0,0 +1,233 @@
+""" 
+@model
+@datasync
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+}
+```
+
+The `@datasync` annotation ensures consistency of your data and gives you delta queries.
+
+> **NOTE**: The `@datasync` annotation is a shorthand for two annotations `@versioned` and `@delta`. Refer to the last section for details.
+
+`@datasync` transforms your model by adding two fields: `updatedAt` and `createdAt`
+
+```graphql
+""" 
+@model
+@datasync 
+"""
+type Comment {
+  id: ID!
+  text: String
+  description: String
+  createdAt: String
+  updatedAt: String
+}
+```
+
+The `@datasync` annotation adds a `sync` query or a delta query:
+```graphql
+type Query {
+  syncComments(lastSync: String!, filter: CommentFilter): CommentDeltaList!
+}
+```
+
+This allows you to get all the changed documents(updated and deleted) in a collection since the `lastSync` timestamp. Internally this uses the `updatedAt` database field to check if any documents in the database have been modified, by comparing client provided `lastSync` timestamp value.
+
+> **NOTE**: The `sync` query also accepts a filter argument for filtering
+
+The `@datasync` annotation also adds a `Delta` type and a `DeltaList` type:
+```graphql
+type CommentDelta {
+  id: ID!
+  text: String
+  description: String
+  createdAt: String
+  updatedAt: String
+  _deleted: Boolean
+}
+
+type CommentDeltaList {
+  items: [CommentDelta]!
+  lastSync: String
+}
+```
+
+The `Delta` type for a model consists of all of the model's transformed properties, as well as a `_deleted` field which is used internally to mark documents as deleted in the database. Thus `delete` mutations only mark documents with `_deleted: true` instead of actually removing them.
+
+The `DeltaList` is a container for `Delta` type, which also returns a `lastSync` timestamp, that can be used in a subsequent `sync` query.
+
+> **NOTE**: `Soft Deletes` strategy can only get you the latest version of changed documents ignoring any in-between states that may have transpired between `lastSync` and now.
+
+### Modify the template to support Data Synchronization
+
+In the [`src/index.ts`](https://github.com/aerogear/graphback/blob/master/templates/ts-apollo-mongodb-backend/src/index.ts) file of the template, use  `DataSyncPlugin` and compliant data sources in `buildGraphbackAPI`:
+
+```typescript
+import { createDataSyncMongoDbProvider, createDataSyncCRUDService, DataSyncPlugin } from '@graphback/datasync'
+
+
+const { typeDefs, resolvers, contextCreator } = buildGraphbackAPI(modelDefs, {
+    dataProviderCreator: createDataSyncMongoDbProvider(db),
+    serviceCreator: createDataSyncCRUDService({ pubSub: new PubSub() }),
+    plugins: [
+        new DataSyncPlugin()
+    ]
+});
+```
+The data sources provided by `@graphback/datasync` ensure that the documents are only marked with `_deleted: true` instead of being removed from the database, meaning that a compatible client can issue a `sync` query to be get info about this deletion.
```

Comment:
```suggestion
The data sources provided by `@graphback/datasync` ensure that the documents are only marked with `_deleted: true` instead of being removed from the database, meaning that a compatible client can issue a `sync` query to obtain information about this deletion.
```